packages/js/product-editor/src/hooks/use-product-manager/use-product-manager.ts

```diff
@@ -18,21 +18,26 @@ export function errorHandler( error: WPError, productStatus: ProductStatus ) {
 		return error;
 	}
 
+	const errorMessageEntry = Object.entries( error ).find(
+		( [ , value ] ) => value !== undefined
+	) as [ string, unknown ] | undefined;
+
+	const errorMessage =
+		errorMessageEntry && typeof errorMessageEntry[ 1 ] === 'object'
+			? { validatorId: errorMessageEntry[ 0 ], ...errorMessageEntry[ 1 ] }
```

Comment:
If you need the validatorId as part of the map value, why not adding it when creating the validation you have access there too. With that this functionality is not needed since the error definition will look like 
Before:
```
error: { validatorId: { ...otherErrorProps: Record } };
```
After:
```
error: { validatorId: { validatorId: string; ...otherErrorProps: Record } };
```

Reply:
Done in commits [4e9a079](https://github.com/woocommerce/woocommerce/pull/49242/commits/4e9a079e71e1cc688135f15d9807d86b6a0f1122) and [2d358a8](https://github.com/woocommerce/woocommerce/pull/49242/commits/2d358a8a165be85aca25bd2d9371648e57e690e4)